docs/development.rst

```diff
@@ -493,48 +581,123 @@ The demographic model should include the following:
-generations), or be able to be converted to physical units using, e.g., mutation rates
-used in the published study.
 
-Note that it is not necessary that the demographic model is attached to a particular
-species. `Stdpopsim` contains a collection of generic models that are widely used in
-developing and testing inference methods. If there is a generic model that does not
-currently exist in our catalog but would be useful to include, we also welcome those
-contributions. Again, you should provide a citation for a generic models, or it
-should be commonly used .
+We can check that the documentation builds properly after implementation by running
+``make`` in the docs directory and opening the Catalog page from the ``docs/_build/``
+directory. See `Documentation`_ for more details.
+
+
+----------------------
+Test the model locally
+----------------------
+
+Once you have written the demographic model function, you should test the model locally
+with ``stdpopsim``. Follow the development `Installation`_ instructions to install the
+development ``stdpopsim`` version along with the requirements.
+
+Now check that your new demographic model function has been imported:
+
+.. code-block:: python
+
+    import stdpopsim
+    species = stdpopsim.get_species("HomSap")
+    for x in species.demographic_models:
+        print(x.id)
+
+    # OutOfAfrica_3G09
+    # OutOfAfrica_2T12
+    # Africa_1T12
+    # AmericanAdmixture_4B11
+    # OutOfAfricaArchaicAdmixture_5R19
+    # Zigzag_1S14
+    # AncientEurasia_9K19
+    # PapuansOutOfAfrica_10J19
+
+
+The example above lists the imported demographic models for humans.
+You should substitute ``"HomSap"`` for which ever species you added your model to.
+Your new model should be printed along with currently available demographic models.
+
+.. note::
+
+    If your demographic model does not print, check:
+
+    - After defining your model function, did you include the call ``_species.add_demographic_model(_model_func_name())``,
+      where ``_model_func_name()`` is your model function name?
+    - Did you install ``stdpopsim`` in your development environment?
+
+    If you are still having trouble, check the
+    `GitHub issues <https://github.com/popsim-consortium/stdpopsim/issues?q=is%3Aissue+adding+demographic+model+>`_,
+    or `open an issue <https://github.com/popsim-consortium/stdpopsim/issues/new>`_.
+
+Next, check that you can successfully run a simulation with your new model with the
+Python API. For example:
+
+.. code-block:: python
+
+    import stdpopsim
+    species = stdpopsim.get_species("HomSap")
+    contig = species.get_contig("chr22", length_multiplier=0.01)
+    model = species.get_demographic_model('OutOfAfrica_3G09')
+    samples = model.get_samples(10, 10, 10)
+    engine = stdpopsim.get_engine('msprime')
+    ts = engine.simulate(model, contig, samples)
+    print(model.num_populations)
+    print([pop.id for pop in model.populations])
+    print(ts.num_sites)
```

Comment:
Maybe this could just refer to other parts of the documentation, to avoid duplication?